### DISCUSSION TITLE
Verbindung openDTU <-> Wechselrichter instabil

### DISCUSSION THREAD

**Discussion by Fokker-1** [tbnobody/OpenDTU#2106] [Q&A]
Hallo,

ich habe eine Fusion OpenDTU mit 2 WR, funktioniert soweit prima.

Die DTU ist im Treppenhaus im Kelleraufgang neben der FritzBox.
Ein WR (2,4 Ghz) ist in der 1. Etage (Entfernung also über 2 Etagen im Haus) da ist die Verbindung stabil.

Der andere WR (868 MHz) ist im Dachgeschoss  (Entfernung also über 3 Etagen im Haus), da kommt es immer wieder zu Verbindungssausfällen und die Web-Anzeige wechselt von blau auf rot.

Durch Veränderung der Position der DTU (aber weiterhin im Treppenhaus im Kellerbereich) und deren Antennen lässt sich zwar eine deutliche Verbesserung erreichen, aber eine immer stabile Verbindung DTU <-> WR kann ich damit nicht sicherstellen.

Das Suchen nach den optimalen Positionen ist äußerst mühsam, da ich keinen direkten Indikator dazu kenne, außer man wartet und beobachtet nach jeder Änderung, ob und wann die Web-Überfläche wieder von blau auf rot und zurück wechselt.

Daher die Fragen:

1) Gibt es in der DTU eine Möglichkeit, die Qualität der Funkverbindungen (Qualität, RSSI) zwischen DTU und WR direkt angezeigt zu bekommen? Das wäre dann ähnlich wie bei der Verbindung DTU <-> FritzBox.

2) Kann man die Sendeleistung von der DTU zum WR in der DTU erhöhen?

Zu diesen beiden Fragen habe ich keine Infos gefunden.

Danke im Voraus und Grüße
Werner

**Comment by ghost** [comment]


**Reply by Fokker-1** [reply]
Wow, markiger Schreibstil.
Den kommentiere ich  nicht.

**Comment by Solarteur** [comment]
Icxh würde die DTU versuchsweise statt im Keller so etwa "mittig" zwischen den WR und dem Router testen. Wenn der Empfang dann stabil ist, liegt die Lösung nahe, auch wenn dieser Standort "optisch" vielleicht nicht besonders schön ist. Die Reichweite der Funkmodule in den WR ist begrenzt.

**Comment by mroenne2022** [comment]
Ich würde auch empfehlen die DTU dichter an die WR postieren und weiter weg vom Router,  so daß ggf beide Verbindungen klappen. Mein Router ist im Keller, die DTU im EG und 3 HM WR sind im Aussenbereich verteilt in 2m Höhe,  Verbindungen klappen 👍

**Comment by Fokker-1** [comment]
Hallo Solateur und mroenne2022,

entschuldigt bitte, dass ich mich erste jetzt wieder melde, aber meine Frau und mich hat Corona voll erwischt, mehrere Tage 39 Grad Fieber ...
....

Ja, ist natürlich absolut richtig, was Ihr schreibt, denn wenn man die DTU vom KG z. B. ins EG versetzt, funktiniert alles problemlos (hatte ich zuvor auch getestet).

Die Sendeleitungen der beden HF-Teile hatte ich zuvor natürlich auch schon zwischen min und max geändert, aber ohne jeden Effekt.
Klar auch, diese Einstellungen wirken wohl nur in Richtung DTU => WR.

Aber es gibt für mich Gründe, das Verlegen der DTU ins EG wirklich nur im "Notfall" zu machen.
Wie Ihr auch schon geschrieben habt, ist die Optik des schwazen Teils mit den 3 Antennen ja nicht so der Hit im Wohnraum.

Aber noch wichtiger ist mir Folgendes:
- Ich betreibe neben der DTU noch 2 Tasmota-Module, alle versorge ich mit den 5 V aus einem USB-Port an der FB.
   Falls ich die Module verlege, müsste ich zusätzliche Steckernetzteile verwenden, was ich aber nicht möchte.

- Ich brauche auch keine permanent laufenden Module, daher schalte ich die ab und nur gelegentlich mal zur Kontrolle ein. 
   Früher musste ich dazu immer ins KG zur FB rennen, da hatte ich einfache, mechanische Schalter für die 5 V jedes Moduls montiert.
    Das habe ich kürzlich geändert und erledige die Ein- / Ausschaltung jedes Moduls nun durch einen Tasmote Relais-Switch über Handy oder PC.

Ich habe daher nun weiter versucht, die Position der DTU im KG und deren Antennausrichtungen zu optimieren. Die Schwierigkeit ist dabei weiterhin, dass ich keinen Indikator habe, der die aktuelle Verbindungsqualität anzeigt (etwa wie bei DTU / FB), man stochert da ziemlich im Dunkeln.

Natürlich weiß ich auch, dass es eine Konsole gibt, @thnobody hatte darauf ja schon in meinem letzten Post hingewiesen. Die Daten dort habe ich dann auch gecheckt, aber, wie schon mehrfach geschrieben, sind meine SW-Kenntnisse absolut minimalistisch, da komme ich nicht weiter.

Ich habe dann nach try-and-error gefummelt - und hurra, ich habe im KG eine Stelle gefunden, an der jetzt die Verbindung klappt. Die Reserven sind da aber wohl nicht üppig, denn bereits leichtes Ändern der Antenneposition führt wieder zu Verbindungsausfällen. Aber egal, ich habe ja eine (mühsame) Lösung.

Trotzdem wäre es natürlich super, wenn die Profis hier die Empfangsqualität in der DTU der Verbindung WR => DTU ausgeben könnten. Schon mal ein großes Danke dafür im Voraus!

Mein Konzept ist nun entsprehend meinen Vorstellungen realisiert:

- ich kann remote alle Module ein- / ausschalten
- sie 5 V-Versorgun erfolgt für alle Module auschließlich aus einem USB-Port der FB
- die DTU verbleibt im KG
- ich kann darüber hinaus auch mein WLAN in der FB remote ein- / ausschalten. Wenn niemand zu Hause ist, braucht es auch kein WLAN.

Somit kann ich jetzt alles auch aus der Ferne ein- / ausschalten. 

Noch mal danke für die freundliche und geduldige Unterstützung.

Viele Grüße
Werner

**Reply by mroenne2022** [reply]
Oha - OK da sind dann bei dir noch ein paar mehr Gründe und Bedingungen, aber dann hast du es jetzt ja einigermassen gelöst. Ich hab meine DTU im schwarzen kleinen Gehäuse mit einer ext. Antenne fürs NRF24 hinterm TV im WoZi stehen , fast nicht zu sehen, von daher ist es für uns kein Problem und vor allem erreiche ich die 3 WR so problemlos.

**Comment by Fokker-1** [comment]
Hallo,

ich melde mich noch einmal - mit Erfreulichem!
Ich habe in den Daten der Konsole nun doch noch wesentliche Hinweise zum Empfangspegel bei 868 MHz (HMS-1000) gefunden.

Ich habe dazu ein Log nach Word kopiert und dort mit der Suchfunktion im Datenwust nach Stichwörtern gesucht, von denen ich annahm. dass diese im fraglichen Zusammenhang stehen könnten (vom Code habe ich ja leider keine Ahnung) - das war dann nach einigen Versuchen auch erfolgreich.

Hier mal ein Ausschnitt:

17:31:14.073 > **Fetch inverter: 114… (HMS-1000)**
17:31:14.138 > TX RealTimeRunData 868.00 MHz --> 15 A0 06 E4 9E 80 16 87 48 80 0B 00 66 86 C0 41 00 00 00 00 00 00 00 00 67 A7 BA 
17:31:14.151 > Interrupt received
17:31:14.170 > **RX 868.00 MHz** --> 95 A0 06 E4 9E 80 16 87 48 01 00 01 01 A9 01 AA 00 E8 00 E8 03 DB 03 DE 00 00 16 | -62 dBm
17:31:14.188 > Interrupt received
17:31:14.200 > RX 868.00 MHz --> 95 A0 06 E4 9E 80 16 87 48 02 BE 3C 00 00 BC 92 06 5A 06 57 09 61 13 8E 07 4D 0C | -62 dBm
17:31:14.206 > Interrupt received
17:31:14.229 > RX 868.00 MHz --> 95 A0 06 E4 9E 80 16 87 48 83 00 00 00 4D 03 E8 01 77 00 0D D2 AB 37 | -62 dBm
17:31:14.352 > RX Period End
17:31:14.352 > Success

"Fetch inverter" ruft  den HMS-1000 auf.
Die hier original augegebene SN habe ich im Beispile durch die Typbezeichnung ersetzt.

Mit "RX 868.00 MHz" wird die korrekte Frequenz angegeben.

Der entscheidende, gesuchte Wert sind dann die **"-62 dBM"**

Ganz ursprünglich lag dieser Wert bei ca. -76 dBm.  Nach weiteren Optimierungen habe ich jetzt einen Wert von ca. -58 dBm.
Bei diesem finalen Zustand konnte ich die DTU auch wieder - wie ursprünglich - im Treppenaufgang im UG neben der FB platzieren!
Seit 3 Tagen nun läuft alles absolut super ohne Verbindungsabbrüche.

Einen wesentlichen Beitrag zur Verbesserung brachte die hier an anderer Stelle schon mal genannte Methode mit der zusätzlichen "Antenne" am WR (danke für diesen Hinweis), das sieht bei mir jetzt so aus:

![Antenne-2_20240704](https://github.com/tbnobody/OpenDTU/assets/140426500/a56fc6e3-abd8-4e5f-bdf2-571103c0f990)

![Antenne-1_20240704](https://github.com/tbnobody/OpenDTU/assets/140426500/0aa4b93f-fff2-43cc-bfdf-095e5cf8af41)

Bei der Länge der "Antenne" habe ich mit 1/4, 1/2,  und 1 Lambda experimentiert, wobei hier 1 Lambda (ca. 34 cm) die besten Ergebniss brachte. Auch die horizontale Ausrichtung hatte merklichen Einfluss.

====

Bei dem HM-800 habe ich z. B. folgendes Log:

17:31:16.944 > **Fetch inverter: 114 … (HM-800)**
17:31:17.844 > TX RealTimeRunData Channel: 75 --> 15 83 14 79 71 80 16 87 48 80 0B 00 66 86 C0 44 00 00 00 00 00 00 00 00 37 98 93 
17:31:17.855 > Interrupt received
17:31:17.864 > RX Channel: 23 --> 95 83 14 79 71 83 14 79 71 02 C6 91 00 04 CA D0 04 88 04 82 09 63 13 8D 05 06 23 | -80 dBm
17:31:17.876 > Interrupt received
17:31:17.887 > RX Channel: 23 --> 95 83 14 79 71 83 14 79 71 83 00 01 00 36 03 E8 01 3F 00 01 33 41 87 | -80 dBm
17:31:17.898 > RX Period End
17:31:17.898 > Middle missing
17:31:17.898 > Request retransmit: 1
17:31:17.898 > TX RequestFrame Channel: 3 --> 15 83 14 79 71 80 16 87 48 81 52 
17:31:17.910 > Interrupt received
17:31:17.921 > RX Channel: 40 --> 95 83 14 79 71 83 14 79 71 01 00 01 01 33 00 DC 02 A4 01 34 00 DB 02 A3 00 04 96 | -80 dBm
17:31:17.937 > RX Period End
17:31:17.937 > Success

Hier ist erstaunlich, dass immer (!) - egal welche Veränderungen ich vornehme - der identische Wert -80 dBm angezeigt wird.
Hat jemand hierzu eine Idee - schon mal danke im Voraus.

Gruß
Werner

**Reply by Fokker-1** [reply]
Hallo schneeer,

danke für die Hinweise.
Nein, die "Antennenerweiterung" hatte keinerlei erkennbare Auswirkungen.

Gruß
Werner

**Comment by tbnobody** [comment]
> Hier ist erstaunlich, dass immer (!) - egal welche Veränderungen ich vornehme - der identische Wert -80 dBm angezeigt wird.

Das nRF 24 Modul kann nur 2 stati anzeigen... entweder < -64dbm oder > -64dbm

Ich habe daher einfach 2 Werte für die Ausgabe gewählt, entweder -30 oder -80. Aber generell ist hardwarebedingt -64dbm die grenze.

**Reply by tbnobody** [reply]
Klar, aber meine internen datenstrukturen erwarten eine zahl als rssi.... "weniger als -64dbm" ist leider keine zahl. genauso wenig wie "<-64dmb"

**Comment by Fokker-1** [comment]
Hallo tbnobody,

danke für die Erläuterungen zu den beiden dBm-Werten, dann ist das jetzt klar.

Ich habe daraufhin noch mal einige Versuche gemacht und speziell die DTU so weit wie möglich an den WR gebracht.
Auch dann kommt der Indikator -80 dBm.

Nach insgesamt ca. 50x  -80 dBm kam dann doch auch 1x  -30 dBm, da braucht es doch wohl schon ziemlich kräftige Werte.

Das habe ich als Information dazu noch gefunden:

![NRF24_1_20240706](https://github.com/tbnobody/OpenDTU/assets/140426500/7c860e8b-5b7b-41c0-ae22-ebe8af98af54)

![NRF24-2_20240706](https://github.com/tbnobody/OpenDTU/assets/140426500/03a45743-aaba-4fb5-8c7a-141d142bba27)

Gibt es eigentlich ein anderes, ggf. alternativ hier einsetzbares Modul, das "echte" dBm-Werte liefert?

Vielen Dank und Grüße
Werner
